nncf/quantization/layers.py

```diff
@@ -43,18 +46,39 @@ def __init__(self, bits=8,
                  signedness_to_force=None,
                  per_channel=False,
                  input_shape=None,
-                 is_weights=False):
+                 is_weights=False,
+                 exponent_bits=0,
+                 mantissa_bits=0,
+                 block_size=0,
+                 folded=None
+                 ):
         self.bits = bits
         self.mode = mode
         self.signedness_to_force = signedness_to_force
         self.per_channel = per_channel
         self.is_weights = is_weights
         self.input_shape = input_shape
+        self.bits = bits if mode != QuantizationMode.BLOCKFP else 0
+        self.exponent_bits = exponent_bits if mode == QuantizationMode.BLOCKFP else 0
+        self.mantissa_bits = mantissa_bits if mode == QuantizationMode.BLOCKFP else 0
+        self.block_size = block_size if mode == QuantizationMode.BLOCKFP else 0
+        self.folded = folded
```

Comment:
Are you happy if I add class IntNQuantizerConfig(QuantizerConfig): and class BFPQuantizerConfig(QuantizerConfig)? bfp does not use some of the INT-N fields so there should be subclasses for both?

Reply:
Right, subclasses for both is what I meant

Reply:
Is it ok to have a factory method in the base class QuantizerConfig.create(mode, ...) that delegates creation of the config to the subclasses - seems tidier passing a mode in rather than switching in the calling function between two constructors.

Reply:
A factory approach seems to be natural in this case, I agree.

Reply:
This is getting a little tangled in the Constraints - where there are iterators over the configs extracting the number of bits (presumably related to HAWQ). I may have to think about this for a while. Any chance I can push that refactor to another pull request in the near future?

Reply:
Ok, created a #140 for this